XmlFormatter/src/Windows/Settings.cs

```diff
@@ -140,40 +149,41 @@ private void importSettingsToolStripMenuItem_Click(object sender, EventArgs e)
+            ISettingScope scope = settingManager.GetScope("Default");
+            if (scope == null)
             {
-                XDocument settingsDocument = XDocument.Load(openFileDialog.FileName);
-                XNode versionNode = settingsDocument.XPathSelectElement("//userSettings//setting[@name='ApplicationVersion']//value");
-                XElement Version = XElement.Parse(versionNode.ToString());
-                string loadedVersion = Version.Value;
-                VersionManager versionManager = new VersionManager();
-                Version importFileVersion = versionManager.ConvertInnerFormatToProperVersion(loadedVersion);
-                if (importFileVersion != versionManager.GetApplicationVersion())
-                {
-                    string message = "Application version is not matching settings version";
-                    message += "\r\n\r\nApplication Version: " + versionManager.GetApplicationVersion();
-                    message += "\r\nSettings Version: " + importFileVersion;
-                    MessageBox.Show(message, "Version not matching", MessageBoxButtons.OK, MessageBoxIcon.Error);
-                    return;
-                }
-                
-                var elements = settingsDocument.XPathSelectElements("//userSettings//setting");
-                
-                foreach (var setting in elements)
-                {
-                    string name = setting.Attribute("name").Value.ToString();
-                    string value = setting.XPathSelectElement("value").FirstNode.ToString();
-                    Type currentType = Properties.Settings.Default[name].GetType();
-                    var realValue = Convert.ChangeType(value, currentType);
-                    Properties.Settings.Default[name] = realValue;
-                }
-                SetupControls();
+                return;
             }
-            catch (Exception ex)
+
+            string storedVersion = scope.GetSetting("ApplicationVersion").GetValue<string>();
+            VersionManager versionManager = new VersionManager();
+
+            string lowestVersion = Properties.Settings.Default.LowestSupportedVersion;
+
+            Version settingVersion = versionManager.ConvertInnerFormatToProperVersion(storedVersion);
+            Version lowVersion = versionManager.ConvertInnerFormatToProperVersion(lowestVersion);
+            Version highVersion = versionManager.GetApplicationVersion();
+
+            if (settingVersion < lowVersion || settingVersion > highVersion)
             {
-                MessageBox.Show(ex.Message, "Import went wrong!", MessageBoxButtons.OK, MessageBoxIcon.Error);
+                string message = "Import failed because setting file is not supported in this version";
+                message += "\r\n\r\nLowest supported version: " + lowVersion.ToString();
+                message += "\r\nApplication version: " + highVersion.ToString();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [There's no need to call 'ToString()', the compiler will do it for you.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5052706)